### PR TITLE
ZK-3749: ListElResolver does not support standard EL syntax

### PR DESCRIPTION
**Pull request by klyvechen**

### CODE REVIEW

File: zkdoc/release-note [hunk 1]
```diff
@@ -58,6 +58,7 @@ ZK 8.5.1
   ZK-3805: Websocket NPE when re-directing or refreshing page
   ZK-3810: Localization doesn't work using ZK 8.5 with WebSockets
   ZK-3743: NPEs while using two event queues with Comet server push
+  ZK-3749: ListElResolver does not support standard EL syntax
```
Comment (scribetw) [inline zkdoc/release-note]: This should be in 8.5.2

File: zktest/test/java/org/zkoss/zktest/zats/test2/B85_ZK_3749Test.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import org.zkoss.zats.mimic.DesktopAgent;
+import org.zkoss.zktest.zats.ZATSTestCase;
+
+public class B85_ZK_3749Test extends ZATSTestCase {
+	@Test
+	public void test() throws Exception {
+		try {
+			DesktopAgent desktop = connect();
+		} catch (Exception e) {
+			Assert.assertFalse(true);
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/B85_ZK_3749Test.java]: Use `Assert.fail()` instead

File: zktest/src/archive/test2/B85-ZK-3749.zul [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                Fri Jan 26 6:36 PM:42 CST 2018, Created by klyve
+
+Copyright (C) 2018 Potix Corporation. All Rights Reserved.
+
+-->
+
+<zk>
+    <zscript><![CDATA[
+
+    ]]></zscript>
```
Comment (scribetw) [inline zktest/src/archive/test2/B85-ZK-3749.zul]: Just delete empty `<zscript>`

File: zktest/src/org/zkoss/zktest/test2/B85_ZK_3749_ViewModel.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import org.zkoss.bind.annotation.ContextParam;
+import org.zkoss.bind.annotation.ContextType;
+import org.zkoss.bind.annotation.Init;
+import org.zkoss.zk.ui.Page;
+import org.zkoss.zul.ListModelArray;
+import org.zkoss.zul.ListModelList;
+
+import java.util.ArrayList;
+import java.util.List;
```
Comment (scribetw) [inline zktest/src/org/zkoss/zktest/test2/B85_ZK_3749_ViewModel.java]: So many unused imports

File: zktest/src/org/zkoss/zktest/test2/B85_ZK_3749_ViewModel.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+public class B85_ZK_3749_ViewModel {
+
+	private ListModelList<String> list;
+
+	@Init
+	public void init() {
+		list = new ListModelList(
```
Comment (scribetw) [inline zktest/src/org/zkoss/zktest/test2/B85_ZK_3749_ViewModel.java]: `new ListModelList<String>`

File: zkbind/src/org/zkoss/bind/xel/zel/ListModelELResolver.java [hunk 6]
```diff
@@ -41,12 +44,29 @@ public Object getValue(ELContext context, Object base, Object property)
...
+					try {
+						property = Fields.get(base, (String) property);
+					} catch (NoSuchMethodException e) {
+
+					}
+					idx = coerce(property);
+				}
+				if (idx == null) {
+					return null;
+				}
```
Comment (scribetw) [inline zkbind/src/org/zkoss/bind/xel/zel/ListModelELResolver.java]: A workaround in `ListModelELResolver` is absolutely wrong. In fact, `ListELResolver` is wrong (https://issues.apache.org/jira/browse/EL-3). If the property is not a number, it should just return null (and not throw an exception) and let `BeanELResolver` handle it.

File: zkdoc/release-note [hunk 7]
```diff
@@ -26,6 +26,7 @@ Apr 10, 2018
   ZK-3916: Automatic session timeout redirects unreliable with websockets enabled
   ZK-3915: Automatic session timeout causes exceptions with websockets enabled
   ZK-3878: missing import org.zkoss.zk.ui.sys in zkex osgi bundle
+  ZK-3749: ListElResolver does not support standard EL syntax
```
Comment (wenninghsu) [inline zkdoc/release-note]: should be in 8.5.2

**Pull request merged**